### ISSUE TITLE
Automatic light/dark theme switching with `prefers-color-scheme` CSS media query

### ISSUE THREAD

**Issue by nekohayo** [moinwiki/moin#1502]
It would be nice if MoinMoin's default stylesheet / theming system would automatically offer a dark stylesheet at nighttime. GitHub itself does this, for example.

My operating system does the switch for me everyday at the correct times, and my web browser follows suit, telling webpages' CSS mediaqueries that dark is now "preferred" or not.

MoinMoin could either automatically offer a dark stylesheet at nighttime using this trick, i.e. "automatic" light and dark themes "based on browser/OS setting", as the default behavior (rather than needing users to turn this on as a setting), though maybe some users will want the ability to override the automatic behavior.

You can detect the system's state using this simple piece of CSS, such as:

`@media (prefers-color-scheme: dark) { blah blah }`

This might be interesting not only for desktop computers, but also for people using the web interface on mobile devices; that way if they're reading messages/directions at night on the street or in a dimly lit location, they won't get blinded by the white screen. It also tends to consume less power on AMOLED displays.

**Comment by omjamnekar** [comment]
@UlrichB22  i would like to work on this

**Comment by UlrichB22** [comment]
@omjamnekar, thanks for your support. I like the new dark theme.

Can you check the other themes? In the “Focus” theme e.g., the input fields are now black, but everything else seems unaffected by the changes. This is a problem with the use of variable names in the different designs. Some variables are used globally. Do you have any ideas for a naming concept that includes global variables and theme-dependent variables?

----

<img width="794" height="539" alt="Image" src="https://github.com/user-attachments/assets/6a75b463-6881-494c-b715-0ce54ecb3380" />

**Comment by UlrichB22** [comment]
@omjamnekar are you still working on this issue?

The PR still needs some optimization, such as 
- using the newly introduced variable names in the common.css file and 
- fixing side effects on themes other than the default “topside” theme.

**Comment by omjamnekar** [comment]
I might not be able to work on it

**Comment by UlrichB22** [comment]
> I might not be able to work on it

Thanks for supporting moin. I'll try to finish the feature.

**Comment by srikaaviya** [comment]
Shall I work on this?

**Comment by UlrichB22** [comment]
@srikaaviya, yes, you're welcome. Please share your ideas with us before you start coding.

The basis is the “topside” theme. I've noticed that the code highlighting features are ugly and buggy in dark mode. You can test with `'help-en/OtherTextItems'` and switch to dark mode in the developer mode of your browser.

<img width="670" height="540" alt="Image" src="https://github.com/user-attachments/assets/0600fb7a-1451-4aee-83fa-88c8ab20731f" />


One possible solution is to use a dark CSS theme generated from Pygments ( e.g. https://github.com/UlrichB22/xstatic-pygments/blob/upd_2192/xstatic/pkg/pygments/data/css/stata-dark.css ) and wrap it into a `'@media (prefers-color-scheme: dark) {}'` code block.

When theme topside is finished we can switch to a generic solution that is tested for all themes.

**Comment by srikaaviya** [comment]
@UlrichB22 , Thank you! So the Topside theme already has the baseline dark mode support. As you suggested, I will focus on:

Using the strata-dark.css styles you linked.
Wrap them in a '@media (prefers-color-scheme: dark)' block.
Add this to theme.css to ensure code blocks are readable in dark mode.

**Comment by UlrichB22** [comment]
@srikaaviya, thanks for your fix. The python code looks much better now:

<img width="795" height="635" alt="Image" src="https://github.com/user-attachments/assets/4b9eda1e-a3c7-477b-8875-18bc0a2b2405" />

There are still some elements with a wrong background-color in dark mode. Theses colors are set in colorful.css, e.g.:

```
.s1 { background-color: #fff0f0 } /* Literal.String.Single */
```
IMO we should set background-color to transparent for those classes in dark mode. Do you want to fix this?

**Comment by srikaaviya** [comment]
@UlrichB22 Yes, I would fix this.

I checked (colorful.css) and found classes that have light background colors.

I will add `background-color: transparent` for those classes inside the existing `@media (prefers-color-scheme: dark)` block, so dark page background shows through instead of the bright patches.

I will change and raise PR.
